lib/Myriad/Subscription/Implementation/Redis.pm

```diff
@@ -114,59 +114,53 @@ async method create_group($receiver) {
+                } catch ($e) {
+                    $log->warnf('skipped subscription on stream %s because: %s will try again', $stream, $e);
+                    return;
+                }
+                await $sink->unblocked;
+                my @events = await $redis->read_from_stream(
+                    stream => $stream,
+                    group  => $group_name,
+                    client => $client_id
                 );
-            } catch {
-                $log->tracef('skipped stream %s because sink is blocked', $stream);
-                next
-            }
-
-            await $self->create_group($item);
 
-            my @events = await $redis->read_from_stream(
-                stream => $stream,
-                group  => $group_name,
-                client => $client_id
-            );
+                for my $event (@events) {
+                    try {
+                        my $event_data = decode_json_utf8($event->{data}->[1]);
+                        $log->tracef('Passing event: %s | from stream: %s to subscription sink: %s', $event_data, $stream, $sink->label);
+                        $sink->source->emit({
+                            data => $event_data
+                        });
+                    } catch($e) {
+                        $log->tracef(
+                            'An error happened while decoding event data for stream %s message: %s, error: %s',
+                            $stream,
+                            $event->{data},
+                            $e
+                        );
+                    }
 
-            for my $event (@events) {
-                try {
-                    my $event_data = decode_json_utf8($event->{data}->[1]);
-                    $log->tracef('Passing event: %s | from stream: %s to subscription sink: %s', $event_data, $stream, $sink->label);
-                    $sink->source->emit({
-                        data => $event_data
-                    });
-                } catch($e) {
-                    $log->tracef(
-                        'An error happened while decoding event data for stream %s message: %s, error: %s',
+                    await $redis->ack(
                         $stream,
-                        $event->{data},
-                        $e
+                        $group_name,
+                        $event->{id}
                     );
                 }
-
-                await $redis->ack(
-                    $stream,
-                    $group_name,
-                    $event->{id}
-                );
             }
-        } else {
-            $log->tracef('No receivers, waiting for a few seconds');
-            await $self->loop->delay_future(after => 5);
-        }
+        }), foreach => [@receivers], concurrent => scalar @receivers);
+    } else {
+        $log->tracef('No receivers, waiting for a few seconds');
+        await $self->loop->delay_future(after => 5);
```

Comment:
There is nothing looping here, maybe we should move it a level above `fmap`

Reply:
yes sure we can do that. However I really think that in fact it's not needed at all.

Because once we start subscription fmap we stay in it and it never finishes. The only case where it will be needed/work is if we don't have any subscription at all, and we add one at runtime.
So I think we are better off without it here, especially that when we reach this place at runtime, we would have already added all subscriptions configured. there is no need to wait.

The wait and trying again is moved to the place where the emitting group is not yet created so we can't create subscription. This one I made the wait to be 1 second, which is im not too sure about